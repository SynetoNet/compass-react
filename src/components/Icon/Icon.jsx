import React from "react";
import PropTypes from "prop-types";
import classNames from "classnames";

const Icon = ({ name, color, size, style, className = "", ...props }) => {
  const classes = classNames({
    [name]: name,
    [`fa-${size}`]: size,
    [className]: className
  });

  return <i className={classes} style={{ color, ...style }} {...props}/>;
};

Icon.propTypes = {
  name: PropTypes.string.isRequired,
  color: PropTypes.string,
  size: PropTypes.oneOf(["xs", "sm", "md", "lg"])
};

Icon.defaultProps = {
  size: "md"
};

export default Icon;
