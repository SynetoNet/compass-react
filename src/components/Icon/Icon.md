#### **Basic usage**

Compass uses [Font Awesome 5.15.4](https://fontawesome.com/icons) for its standard icon set.

All free sets are included:

- [Font-Awesome solid](https://fontawesome.com/icons?d=gallery&s=solid&m=free) (fa / fas)
- [Font-Awesome regular](https://fontawesome.com/icons?d=gallery&s=regular&m=free) (far)
- [Font-Awesome brands](https://fontawesome.com/icons?d=gallery&s=brands&m=free) (fab)

```jsx
<Icon name="fas fa-user" />
<i className="mx-1" />
<Icon name="far fa-user" />
<i className="mx-1" />
<Icon name="fab fa-github" />
```

#### **Sizes**

```jsx
<Icon name="fas fa-user" size="xs"/>
<span className="me-2"/>

<Icon name="fas fa-user" size="sm"/>
<span className="me-2"/>

<Icon name="fas fa-user" size="md"/>
<span className="me-2"/>

<Icon name="fas fa-user" size="lg"/>
```

#### **Colors**

```jsx
<Icon name="fas fa-user" color="#0000FF" />
```

#### **Inherited styles**

Icon component inherits the **`font-size`** and **`color`** styles of their parent:

```jsx
<div style={{ color: "#0000FF", fontSize: "30px" }}>
  <Icon name="fas fa-user" />
</div>
```

#### **Custom styles**

You can pass both **`style`** and **`className`** to customize the Icon:

```jsx
<Icon
  name="fas fa-user"
  style={{ fontSize: "30px" }}
  className="text-success"
/>
```
