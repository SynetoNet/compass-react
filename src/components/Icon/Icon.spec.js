import React    from "react"
import Icon     from "./Icon"
import renderer from "react-test-renderer"

test("renders default Icon", () => {
  const component = renderer.create(<Icon name="fas fa-user"/>)
  expect(component.toJSON()).toMatchSnapshot()
})
