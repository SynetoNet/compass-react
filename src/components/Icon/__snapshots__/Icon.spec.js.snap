// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders default Icon 1`] = `
<i
  className="fas fa-user fa-md"
  style={
    Object {
      "color": undefined,
    }
  }
/>
`;
